Code/Tools/StaticLinkUtil/StaticLinkUtil.cpp

```diff
@@ -723,13 +712,35 @@ class ezStaticLinkerApp : public ezApplication
           // part such that it will reference all the other files
           if (sFileContent.FindSubString("EZ_STATICLINK_LIBRARY"))
           {
+            m_FilesToModify.Insert(sFile);
+
             ezLog::Info("Found macro 'EZ_STATICLINK_LIBRARY' in file '{0}'.", &sFile.GetData()[m_sSearchDir.GetElementCount() + 1]);
 
             if (!m_sRefPointGroupFile.IsEmpty())
               ezLog::Error("The macro 'EZ_STATICLINK_LIBRARY' was already found in file '{0}' before. You cannot have this macro twice in the same library!", m_sRefPointGroupFile);
             else
               m_sRefPointGroupFile = sFile;
           }
+
+          if (sFileContent.FindSubString("EZ_STATICLINK_FILE_DISABLE"))
+            continue;
+
+          m_FilesToModify.Insert(sFile);
+
+          bool bContainsGlobals = false;
+
+          bContainsGlobals = bContainsGlobals || (sFileContent.FindSubString("EZ_STATICLINK_LIBRARY") != nullptr);
```

Comment:
@Ingrater Reflection isn't the only thing that needs to be considered. We have a couple more globals, like CVars, plugin references and the startup blocks. Those are all that I could find atm. Of course now that means we need to keep an eye out for additional globals and make sure they get included as well.